standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -701,6 +705,8 @@ struct UniqueConstraintsRequest {
   1: required string catName,
   2: required string db_name,
   3: required string tbl_name,
+  4: optional string validWriteIdList,
+  5: optional i64 tableId=-1
```

Comment:
I wonder if a common object to reference a table might be handy ; these 5 fields are copied to a bunch of requests/responses - having a common object might make it easier to work with them

Reply:
I meaned something which could retain that we may change the request further but have the table related stuff separated:
{code}
struct UniqueConstraintsRequest {
1: TableReference table;
}

struct TableReference {
 1: required string catName,
  2: required string db_name,
  3: required string tbl_name,
  4: optional string validWriteIdList,
  5: optional i64 tableId=-1
}

{code}

this could also help clean things up a bit

Reply:
@kgyrtkirk I initially though the same. But doing this change will break the metastore api contract and all call from HiveMetaStoreClient.java from older clients will not be compatible any more. So that's why instead of extracting out the common variable to another struct I added in the same as optional.